runtime/src/bank.rs

```diff
@@ -404,29 +404,35 @@ impl CachedExecutors {
         })
     }
     fn put(&mut self, pubkey: &Pubkey, executor: Arc<dyn Executor>) {
-        if !self.executors.contains_key(pubkey) && self.executors.len() >= self.max {
-            let mut least = u64::MAX;
-            let default_key = Pubkey::default();
-            let mut least_key = &default_key;
-
-            for (key, entry) in self.executors.iter() {
-                let count = entry.prev_epoch_count + entry.epoch_count.load(Relaxed);
-                if count < least {
-                    least = count;
-                    least_key = key;
+        let entry = if let Some(entry) = self.executors.get(pubkey) {
```

Comment:
If we `remove()` instead of `get()` here, the entry can be reused and avoid the atomic ops